### PR TITLE
Documentation for https://github.com/zendframework/zf2/pull/6431

### PR DESCRIPTION
**Pull request by brettminnie**
Documentation required for the merge method on the input filter

### CODE REVIEW

**Comment by GeeH** [comment]
Any comments @Ocramius? I haven't kept up with this feature.

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 1]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+            public function __construct()
+            {
+                $this->inputFactory = new InputFactory();
+                $this->setValidators();
+            }
+
+            /**
+             * Loads the validators
+             */
+            protected function setValidators()
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: Please remove this method: we are writing examples, keep them as small as possible.

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 2]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+            {
+                $this->setNameValidator();
+            }
+
+            /**
+             * Creates a validator to check the name property
+             */
+            protected function setNameValidator()
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: Compact this with the constructor

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 3]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+        * Filter to ensure a name property is set and > 8 characters
+        */
+        class NameInputFilter extends InputFilter
+        {
+            /** @var InputFactory */
+            protected $inputFactory;
+
+            public function __construct()
+            {
+                $this->inputFactory = new InputFactory();
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: Don't think this is needed: the input factory is implicitly built internally, no?

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 4]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+                $this->setValidators();
+            }
+
+            /**
+             * Loads the validators
+             */
+            protected function setValidators()
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: Same as above

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 5]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+            {
+                $this->setEmailValidator();
+            }
+
+            /**
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: Same as above

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 6]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+                                    'name'    => 'email_address',
+                                )
+                            )
+                        )
+                    )
+                );
+            }
+        }
+
+        class SimplePerson implements InputFilterAwareInterface
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: Please don't use the `InputFilterAwareInterface` in examples: it's a bad design.

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 7]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+            {
+                $this->email = $email;
+            }
+
+             /**
+              * Retrieve input filter
+              *
+              * @return InputFilterInterface
+              */
+            public function getInputFilter()
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: This method is the actual example: you can actually create a standalone example without the definitions of two input filter classes

File: docs/languages/en/modules/zend.input-filter.intro.rst [hunk 8]
```diff
@@ -167,6 +167,197 @@ appropriate object. You may create either ``Input`` or ``InputFilter`` objects i
...
+                return $this->inputFilter;
+            }
+
+            /**
+             * Set input filter
+             *
+             * @param  InputFilterInterface $inputFilter
+             * @return InputFilterAwareInterface
+             */
+            public function setInputFilter(InputFilterInterface $inputFilter)
```
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: Kill this :-)

File: docs/languages/en/modules/zend.input-filter.intro.rst (hunk 6 above)
Comment (Ocramius) [inline docs/languages/en/modules/zend.input-filter.intro.rst]: I'd actually remove the class altogether, stripping the example to a minimum

**Comment by brettminnie** [comment]
@Ocramius just got back from my break, will go through these comments now and update the pull request shortly

**Comment by Ocramius** [comment]
@brettminnie missed the update. Merging, thanks!

**Pull request merged**